andela/travel_tool_back
# #159725850 Create api to add a new request to the db

#### What does this PR do?
- This PR creates the api endpoint for posting a new request to the database.
#### Description of Task to be completed?
- create `Requests model` and `migrations`
- create middleware to authenticate a user
- create middleware to validate user input
- create route abnd controller to post a new request
- create tests for the api

#### How should this be manually tested?
- clone the repo `git clone https://github.com/andela/travel_tool_back.git`
- run `cd travel_tool_back` to switch to the project directory
- set up `.env` from the example provided in `.env.example`
- run `yarn install` to install dependencies
- run `make start` to start the app
- after you are logged in, visit the developer console in your browser and open the `Application` tab, you should see your `jwt-token` under cookies.
- copy the token and add it to the `authorization` header in `postman`
- in `postman`, make a `POST` request to the endpoint `http://localhost:5000/api/v1/requests` with the following fields `name`, `origin`, `destination`, `manager`, `gender`, `department`, `role`, `arrivalDate`, `departureDate`.

#### Any background context you want to provide?
[![Run in Postman](https://run.pstmn.io/button.svg)](https://app.getpostman.com/run-collection/10782b1e7b69ae08baa4)

You should add the `authorization token` as described in the tests section above

#### Screenshots (if appropriate)
![image](https://user-images.githubusercontent.com/25783086/44529736-b5f09f80-a6e4-11e8-8d69-73b212834410.png)
![image](https://user-images.githubusercontent.com/25783086/44529782-cd2f8d00-a6e4-11e8-9a41-d7fe3bcec1ca.png)


#### What are the relevant pivotal tracker stories?
[#159725850](https://www.pivotaltracker.com/story/show/159725850)
